Chore/sentry errors

tobias-oetzel opened the pull request:
we currently do not get any error logs besides the general error handler, that does not report to sentry because there is no error wrapping

tobias-oetzel committed:
chore: auth directive report to sentry/log

we currently do not get any error logs besides the general error handler, that does not report to sentry because there is no error wrapping

tobias-oetzel committed:
fix tests

pteich commented on directive/authorized.go:
```diff
@@ -55,102 +56,114 @@ func extractNestedKeyFromArgs(args map[string]any, paramName string) (string, er
-			log.Trace().Str("user", userID).Msg("using spiffe user in authorized directive")
-		}
+	if evaluatedEntityType == "" {
+		return nil, fmt.Errorf("make sure to either provide entityType or entityTypeParamName")
```
Consider using an exported error var (or at least not mix `fmt.Errorf` and `errors.New` from above)

pteich commented on directive/authorized.go:
```diff
@@ -55,102 +56,114 @@ func extractNestedKeyFromArgs(args map[string]any, paramName string) (string, er
 
-		token, err := openmfpcontext.GetAuthHeaderFromContext(ctx)
-		hasToken := err == nil
+		return check, err
```
I would explicitly do `return check, nil` as the error is already checked and this is easier to read

pteich reviewed (changes requested):
I had some comments and suggestions. Exporting error vars is optional (could make sense, though)

tobias-oetzel committed:
review comments

extracted a sctruct because there were too many params
separated the actual check from the preparation
reported only client errors to sentry

tobias-oetzel commented on directive/authorized.go:
```diff
@@ -55,102 +56,114 @@ func extractNestedKeyFromArgs(args map[string]any, paramName string) (string, er
-		}
+func executeTheAuthCheck(ctx context.Context, openfgaClient openfgav1.OpenFGAServiceClient, log *logger.Logger, entityParamName string, entityTypeParamName *string, entityType *string, relation string, next graphql.Resolver) (interface{}, error) {
+	if openfgaClient == nil {
+		return nil, errors.New("OpenFGAServiceClient is nil. Cannot process request")
```
i think we will not check those outside of the package. so for now i would skip the export

pteich reviewed (approved):
Approved with comments

tobias-oetzel committed:
review comments 2
